### DISCUSSION TITLE
writeContractAsync function getting failed - using the useWriteContract

### DISCUSSION THREAD

**Discussion by subhradwip01** [wevm/wagmi#4130] [Q&A]
I was trying to send non-native tokens on different chains.. For now, I was testing sepolia test net USDT.. but the writeContractAsync function not working it keeps this error... But even if I add the main net USDT address same issue occurs..
Attaching the codes and issue: (Can not share full code to maintain the confidentiality)
```
import { sepolia } from 'viem/chains';
import { useConnect, useAccount, useWriteContract } from 'wagmi'

 const { writeContractAsync } = useWriteContract()
const data = await writeContractAsync({
        chainId: sepolia.id,
        address: '0xaa8e23fb1079ea71e0a56f48a2aa51851d8433d0', 
        functionName: 'transfer',
        abi: [{ "inputs": [ { "internalType": "address", "name": "recipient", "type": "address" }, { "internalType": "uint256", "name": "amount", "type": "uint256" } ], "name": "transfer", "outputs": [ { "internalType": "bool", "name": "", "type": "bool" } ], "stateMutability": "nonpayable", "type": "function" }],
        args: [
          '0x64498163f2b3E5AA871d335F4CBA6d5b5DcdD6BA',
          price * 1000000,
        ],
      })
```
      
      CONFIGS:
```
import { http, createConfig } from '@wagmi/core'
import { sepolia } from '@wagmi/core/chains'
import { injected } from '@wagmi/connectors'

export const config = createConfig({
  chains: [sepolia],
  connectors: [injected()],
  transports: {
    [sepolia.id]: http(),
  },
  ssr: true, 
})
```
 

ERROR:
![Screenshot 2024-07-11 202156](https://github.com/wevm/wagmi/assets/88609894/34fff00f-46e4-4929-9f65-b757f1895b02)

Please give me the solution. 
I know there is function called sendTransaction but it is only for native coins I guess.. we need to do for all coins and token.. What could be the solution for that

**Comment by flymauntains** [comment]
I need wagmi version. V1 or V2?

**Comment by dev-mata** [comment]
did you fix it? im getting the same error
